internal/pkg/cli/pipeline_status.go

```diff
@@ -19,14 +21,22 @@ type pipelineStatusVars struct {
 type pipelineStatusOpts struct {
 	pipelineStatusVars
 
+	pipelineSvc         pipelineGetter
 	statusDescriber     pipelineStatusDescriber
 	initStatusDescriber func(opts *pipelineStatusOpts) error
 }
 
 func newPipelineStatusOpts(vars pipelineStatusVars) (*pipelineStatusOpts, error) {
-	return &pipelineStatusOpts{
+	session, err := session.NewProvider().Default()
+	if err != nil {
+		return nil, fmt.Errorf("session: %w", err)
+	}
+
+	opts := &pipelineStatusOpts{
 		pipelineStatusVars: vars,
-	}, nil
+		pipelineSvc: codepipeline.New(session),
+	}
+	return opts, nil
```

Comment:
So I missed this in https://github.com/aws/amazon-ecs-cli-v2/pull/954/, but it looks like you're not defining `statusDescriber` or `initStatusDescriber`in the constructor here. Depending on what your pipeline status describer implementation looks like, you may not need the `initStatusDescriber` function at all, i.e. you may be able to set `statusDescriber` by calling the constructor for your pipelineStatusDescriber.